zipkin/src/tracer.cc

```diff
@@ -111,4 +112,51 @@ void Tracer::reportSpan(Span &&span) {
 void Tracer::setReporter(ReporterPtr reporter) {
   reporter_ = std::move(reporter);
 }
+
+uint64_t Tracer::getDroppedSpanCount() const {
+  ReporterImpl *reporter = reinterpret_cast<ReporterImpl *>(reporter_.get());
+  if (reporter) {
+    return reporter->getDroppedSpanCount();
```

Comment:
These are _technically_ race conditions. One could set the reporter to a `nullptr`, which would run the destructor on this pointer. Past that point, accessing the pointer is undefined behavior.

I say let's make it so you can't set the reporter pointer after construction.

Reply:
👍